### ISSUE TITLE
BackendEntrypoint bug chunking data

### ISSUE THREAD

**Issue by timvgl** [pydata/xarray#8810]
### What happened?

Hi together,
I have written a class for the BackendEntrypoint. Since the data can be quite large, I wrote huge parts by making use of the dask.delayed function. As one of the last steps I am redefining the chunks. The code is loading single files and concatenates them into a xarray. As a result, it makes sense to chunk for each single file. This chunking is not forwarded to the main program, where the class is used:
`dset = dset.chunk(dict(zip(dims, list(reversed([list(reversed(list(shape)))[i] if i < 4 else 1 for i in range(len(list(shape)))])))))`
`return dset`
https://github.com/timvgl/mumaxXR/blob/daa4345b482197b65a7bd88df1552d9fa48b8bf7/src/mumaxXR/OvfEngine.py#L356-L357
Between dset = .... and return dset dset.chunksizes gives the expcted values. However, in the main program, the chunking is not available anymore and dset.chunksizes gives Frozen({}).
Thanks for your help!
Tim

### MVCE confirmation

- [ ] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [ ] Complete example — the example is self-contained, including all data and the text of any traceback.
- [ ] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [ ] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [ ] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Environment

xarray version: 2023.1.0
Linux

**Comment by max-sixty** [comment]
Thanks for the issue, but we really need a more focused example to make progress here

**Comment by keewis** [comment]
I agree that debugging the somewhat complex piece of code you linked to is a bit much to ask for an issue (especially given that we're already somewhat behind in answering some issues).

However, glancing over the code, I would point you to the [preferred chunksizes](https://docs.xarray.dev/en/stable/internals/how-to-add-new-backend.html#preferred-chunk-sizes) section in the docs, in particular this sentence:
> Note that the backend is not directly involved in [Dask](https://dask.org/) chunking, because Xarray internally manages chunking.

In other words, your backend is supposed to use the lazy indexing classes mentioned in that document, and you will get the `dask` handling for free. If that doesn't work for you (for whatever reason), remember that having a function separate from `xarray`'s backend machinery is totally fine, as well.

**Comment by timvgl** [comment]
Of course, debugging that code is not what I wanted you to do - you guys have different things to do. Thanks for the hint with the lazy indexing classes. I am gonna check that out. I wrote a shorter example to show that this is a more general problem.
I am not sure whether chunking within the backend making use of the chunking mechanism provided by xarray is supposed to be supported. 
```
import xarray as xr
class TestEngine(xr.backends.BackendEntrypoint):
    def open_dataset(
        self,
        filename_or_obj,
        *,
        drop_variables=None,
    ):
        dataset = xr.open_dataset(filename_or_obj)
        chunkDict = {}
        for dim in dataset.dims:
            chunkDict[dim] = 1
        dataset = dataset.chunk(chunkDict)
        print(dataset.chunksizes)
        return dataset

dataset = xr.open_dataset(path_nc_file, engine=TestEngine)
print(dataset.chunksizes)
```
In this case, chunking from xarray in the backend is not forwarded.
This is a decision you just have to make.
If this is not wanted, this issue can be closed from my side.

**Comment by keewis** [comment]
thanks for the minimal example. That reduces this issue to just "Should we allow backends to return chunked datasets?", which is much easier to discuss.

**Comment by timvgl** [comment]
Yeah I have updated the code already making use of the lazy loading structure. I did use the mf function in the past already. It is just more conviniend if the user doesn't have to define args that are the same always (like parallel=True, concat_dim='t' etc.). There are also some other reasons for using the original function instead of mf, but this would go to far.
